src/placement/pl_map_common.c

```diff
@@ -196,23 +197,28 @@ int
 remap_list_fill(struct pl_map *map, struct daos_obj_md *md,
 		struct daos_obj_shard_md *shard_md, uint32_t r_ver,
 		uint32_t *tgt_id, uint32_t *shard_idx,
-		unsigned int array_size, int myrank, int *idx,
+		unsigned int array_size, int *idx,
 		struct pl_obj_layout *layout, d_list_t *remap_list,
 		bool fill_addition)
 {
 	struct failed_shard     *f_shard;
 	struct pl_obj_shard     *l_shard;
-	int                     rc = 0;
 
 	d_list_for_each_entry(f_shard, remap_list, fs_list) {
 		l_shard = &layout->ol_shards[f_shard->fs_shard_idx];
 
 		if (f_shard->fs_fseq > r_ver)
 			break;
 
+		/**
+		 * NB: due to colrelocation, the data on healthy
+		 * shard might need move to, so let's include
+		 * UPIN status here as well.
+		 */
```

Comment:
Could you improve this comment if possible? Like illustrate the extra data movement caused by colocation problem?

BTW, is this colocation issue being handled by current reintegration code?

Reply:
Hmm, it seems working, but I need further testing here.